### PR TITLE
Nest js login admin

### PR DESCRIPTION
**Pull request by LiamMillerGofore**
This includes all code needed for S1-T4 Nestjs login for admin and includes tests. Information for tests has been provided within backend/Readme.md.

### CODE REVIEW

File: docker-compose.yml [hunk 1]
```diff
@@ -2,7 +2,7 @@ version: "3"
 services:
   backend:
     build: ./backend
-    command: npm run start:dev
+    command: sh -c "npm install && npm run start:dev"
```
Comment (khanguyen1207) [inline docker-compose.yml]: Is it necessary to run `npm install` every time we start the backend service? I'd say running `docker-compose build` when we introduce new dependencies to the project is quite OK.

File: backend/src/authentication/authentication.service.ts [hunk 2]
```diff
@@ -0,0 +1,35 @@
...
+        } as Admin;
+        await this.adminService.createUser(admin);
+        return `${registrationData.email} luotu.`;
+    }
+
+    async loginAdmin(loginData: LoginAdminDto): Promise<any> {
+        const user = await this.adminService.getUser(loginData.email);
+        if (!user) { throw new BadRequestException(); }
+        const passwordMatches = await compare(loginData.password, user.password);
+        if (!passwordMatches) { throw new UnauthorizedException(); }
```
Comment (khanguyen1207) [inline backend/src/authentication/authentication.service.ts]: Injecting a meaningful message would help frontend to tell user why the error happens. Make use of `content.json` is encouraged but not strictly required

File: backend/src/authentication/authentication.consts.ts [hunk 3]
```diff
@@ -0,0 +1,2 @@
+export const saltRounds = 10;
+export const jwt = { secret: 'Remember to make me more secure before prod!', expiry: `7d` };
```
Comment (khanguyen1207) [inline backend/src/authentication/authentication.consts.ts]: This is a potential security bug

File: docker-compose.yml (hunk 1 above)
Comment (LiamMillerGofore) [reply docker-compose.yml]: Good spot, will remove.

File: backend/src/authentication/authentication.consts.ts (hunk 3 above)
Comment (LiamMillerGofore) [reply backend/src/authentication/authentication.consts.ts]: The goal is to move this into environment variables when preparing the environment files for UAT Deploy (S1T1).

Comment (LiamMillerGofore) [reply backend/src/authentication/authentication.consts.ts]: Have resolved this in advance to ensure it is enforced later.

**Pull request merged**